### PR TITLE
Add a missing note to v4.0.9 release notes

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Added an entry [#21290](https://github.com/pingcap/tidb/issues/21290) in the release notes.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-4.0.9.md [hunk 1]
```diff
@@ -155,6 +155,7 @@ TiDB 版本：4.0.9
     - 修复了快照读能够命中 lock cache，返回错误结果的问题 [#21539](https://github.com/pingcap/tidb/pull/21539)
     - 修复了在同一个事务中读取大量数据时可能发生的内存泄漏问题 [#21129](https://github.com/pingcap/tidb/pull/21129)
     - 修复了在子查询中省略表别名时的语法解析错误问题 [#20367](https://github.com/pingcap/tidb/pull/20367)
+    - 修复了在一个事务中更新并查询但返回记录为空的问题[#21290](https://github.com/pingcap/tidb/issues/21290)
```
Comment (qw4990) [inline releases/release-4.0.9.md:158]: 要不用 bugfix PR 中原作者的描述：`修复了 IN 函数当参数为 time 类型时可能返回错误结果的问题`

Comment (TomShawn) [inline releases/release-4.0.9.md:158]: According to https://github.com/pingcap/tidb/issues/21290, should it be "修复了查询中的 `IN` 函数如果包含时间类型参数会导致查询结果为空的问题"？

Comment (hfxsd) [reply releases/release-4.0.9.md]: Done

**Review by xiongjiwei — APPROVED** [review]

File: releases/release-4.0.9.md [hunk 2]
```diff
@@ -155,6 +155,7 @@ TiDB 版本：4.0.9
     - 修复了快照读能够命中 lock cache，返回错误结果的问题 [#21539](https://github.com/pingcap/tidb/pull/21539)
     - 修复了在同一个事务中读取大量数据时可能发生的内存泄漏问题 [#21129](https://github.com/pingcap/tidb/pull/21129)
     - 修复了在子查询中省略表别名时的语法解析错误问题 [#20367](https://github.com/pingcap/tidb/pull/20367)
+    - 修复了 IN 函数当参数为 time 类型时可能返回错误结果的问题 [#21290](https://github.com/pingcap/tidb/issues/21290)
```
Comment (TomShawn) [inline releases/release-4.0.9.md:158]:
```suggestion
    - 修复了查询中 `IN` 函数的参数为 time 类型时可能返回错误结果的问题 [#21290](https://github.com/pingcap/tidb/issues/21290)
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

**Review by shichun-0415 — APPROVED** [review]

**Pull request merged**